src/main/java/org/folio/list/services/ListService.java

```diff
@@ -330,8 +344,15 @@ private String getUserFriendlyQuery(String fqlCriteria, EntityType entityType) {
     return userFriendlyQueryService.getUserFriendlyQuery(fql.fqlCondition(), entityType);
   }
 
-  private EntityType getEntityType(UUID entityTypeId) {
-    return entityTypeClient.getEntityType(entityTypeId);
+  private EntityType getEntityType(UUID entityTypeId, ListActions attemptedAction) {
+    try {
+      return entityTypeClient.getEntityType(entityTypeId);
+    } catch(FeignException.Unauthorized e) {
+      String message = e.getMessage();
+      throw new InsufficientEntityTypePermissionsException(entityTypeId, attemptedAction, message);
+    } catch(FeignException.NotFound e) {
+      throw new NotFoundException("Entity type with id " + entityTypeId + " was not found.");
```

Comment:
Using this exception here to give a nicer error message and code. (The feign error message is pretty nasty, and right now we're just sending back a 500 if the entity type doesn't exist).